src/sagemaker/model_card/model_card.py

```diff
@@ -208,6 +208,257 @@ def search_model_associated_model_cards(model_id: str):
+                            "Make sure the model package name or ARN is valid."
+                        )
+                    )
+                else:
+                    raise
+            return model_package_response
+
+        def search_model_package_associated_model_cards(model_id: str):
+            """Check if a model card already exists for this model package version.
+
+            Args:
+                model_id (str): Model package version ARN.
+            """
+            response = sagemaker_session.sagemaker_client.search(
+                Resource="ModelCard",
+                SearchExpression={
+                    "Filters": [
+                        {
+                            "Name": "ModelId",
+                            "Operator": "Equals",
+                            "Value": model_id,
+                        }
+                    ]
+                },
+            )
+            return [c["ModelCard"]["ModelCardName"] for c in response["Results"]]
+
+        if not sagemaker_session:
+            sagemaker_session = Session()  # pylint: disable=W0106
+
+        model_package_response = call_describe_model_package()
+
+        associated_model_cards = search_model_package_associated_model_cards(model_package_response["ModelPackageArn"])
+        if associated_model_cards:
+            raise ValueError(
+                f"The model package has been associated with {associated_model_cards} model cards."
+            )
+
+        kwargs.update(
+            {
+                "model_package_arn": model_package_response["ModelPackageArn"],
+                "model_package_group_name": model_package_response["ModelPackageGroupName"],
+                "model_package_status": model_package_response["ModelPackageStatus"],
+                "model_approval_status": model_package_response["ModelApprovalStatus"],
+                "model_package_version": model_package_response["ModelPackageVersion"],
+            }
+        )
+
+        for key in OPTIONAL_FIELDS_MAP:
+            if key in model_package_response:
+                kwargs.update(
+                    {
+                        OPTIONAL_FIELDS_MAP[key]: model_package_response[key]
+                    }
+                )
+
+        if "UserProfileName" in model_package_response["CreatedBy"]:
+            model_package_creator = ModelPackageCreator(user_profile_name=model_package_response["CreatedBy"]["UserProfileName"])
+            kwargs.update(
+                {
+                    "created_by": model_package_creator
+                }
+            )
+
+        if "InferenceSpecification" in model_package_response:
+            containers_response = model_package_response["InferenceSpecification"]["Containers"]
+            containers = []
+            for container in containers_response:
+                if ("ModelDataUrl" in container and "NearestModelName" in container):
```

Comment:
we could simplify the code logic to just update the container's arguments to make the code intent clearer 
```
inference_specification_fields_mapping = {"NearestModelName": "nearest_model_name", "ModelDataUrl": "model_data_url"} # to be defined in the class level 

args = {"image": container["Image"]}
for field, arg_key in inference_specification_fields_mapping:
    if field in container:
        args[arg_key] = container[field]
containers.append(Container(**args))
```